src/OrleansCodeGenerator/GrainMethodInvokerGenerator.cs

```diff
@@ -255,7 +255,7 @@ private static MethodDeclarationSyntax GenerateInvokeMethod(Type grainType, Meth
             for (var i = 0; i < methodParameters.Count; i++)
             {
                 var parameter = methodParameters[i];
-                var parameterType = parameter.ParameterType.GetTypeSyntax();
+                var parameterType = parameter.ParameterType.GetTypeSyntaxAsDynamic();
```

Comment:
Maybe we should name this `GetTypeSyntaxUsingDynamic` (or `...DynamicGenerics`?), what do you think? I'm open to other suggestions.